### PR TITLE
Update link in assignment

### PR DESCRIPTION
**Pull request by odilson-dev**
## Because

This PR is about updating the link of Ressources in the asiignment section, it's improve the experience of students by 
guiding them directly to the section we want them to read.

## This PR

- Update the link to the right section

## Issue

Closes #XXXXX

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md
Comment (CouchofTomato) [inline ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md:223]: This should state sections 7, 8, and 21 now.

File: ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md
Comment (CouchofTomato) [inline ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md:224]: We don't want to change the link here. `overriding-conditions` is where we want to link to on the page but what we need to do is change the section numbers to reflect that. From looking at the new sections I think it might just be section 8 now that covers this but we should probably keep 9 as well.

File: ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md
Comment (CouchofTomato) [inline ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md:225]: This is now section 12 so we should update this.

File: ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md
Comment (CouchofTomato) [inline ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md:226]: This is now section 18 so we should update this.

**Review by CouchofTomato — CHANGES_REQUESTED** [review]

**Comment by CouchofTomato** [comment]
You also have some merge conflicts you'll need to address before we can merge this.

File: ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md
Comment (odilsonjs) [reply ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md:223]: Okay

File: ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md
Comment (odilsonjs) [reply ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md:224]: Okay, i replaced #null-relation back to #overriding-conditions. You'll let me know if any changes are required.

File: ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md
Comment (odilsonjs) [reply ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md:225]: Yes, it is actually section 12

File: ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md
Comment (odilsonjs) [reply ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md:226]: Okay

**Comment by CouchofTomato** [comment]
> Hi, i resolved merge conflicts, please let me know if any changes are requirded before merge.

It looks like merging in main may have reset one of the lines because I don't think it should be 21

image

**Comment by odilsoncode** [comment]
> > Hi, i resolved merge conflicts, please let me know if any changes are requirded before merge.
> 
> It looks like merging in main may have reset one of the lines because I don't think it should be 21
> 
> image

Just fixed that. Thanks

File: ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md
Comment (CouchofTomato) [inline ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md:224]: Really sorry mate. I should have picked this up before but I don't think we want sections 9 - 12 here. I think this might be another corrected one from the merge because the next line is for section 12. I'm guessing this should be sections 9 - 11

**Review by CouchofTomato — CHANGES_REQUESTED** [review]
Just the one small change that I missed before. Sorry.

File: ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md
Comment (odilsoncode) [reply ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md:224]: Nevermind mate, please can you check the link in the fourth bull point, it's actually pointing to the section 8.
Normally, i think it should be from section 8 to 11.

Screenshot 2024-01-05 at 10-13-49 Active Record Queries The Odin Project

However, we've already added section 8 in the third bull point🤔

Comment (CouchofTomato) [reply ruby_on_rails/advanced_forms_and_activerecord/active_record_queries.md:224]: Yeah I think point 3 should be section 7 and 21 and then point 4 should be 8 - 11

**Comment by CouchofTomato** [comment]
Hey @odilsoncode 

Just wondering if you were still interested in finishing this PR up?

**Comment by odilsoncode** [comment]
Hi @CouchofTomato 
I thought this PR was already completed and merged but it seems that it is not.
Please, let me know what other changes i've to make again so we can finish up.

**Comment by CouchofTomato** [comment]
Hey @odilsoncode 

There are some merge conflicts. We can't merge until these are resolved.

**Comment by odilsoncode** [comment]
Hmmm... hi @CouchofTomato , i'm currently facing a bunch of errors on this PR, and i'm not sure where they come from. Any assistance please?

**Comment by thatblindgeye** [comment]
@odilsoncode  You can ignore those errors, they're just from our new linter that is helping us flag things to help lessons comply with our layout style guide.

@CouchofTomato do the changes to the PR look good to you?

**Comment by MaoShizhong** [comment]
Is there stuff that's meant to be changed here? Currently the diff only shows a single change that's removing trailing whitespace, but otherwise doesn't seem to show any changes from the current lesson.

**Comment by CouchofTomato** [comment]
Yeah it was supposed to update the chapter links for the exercises because the rails guide changed them. I think I'll have to just edit it at some point.

**Comment by MaoShizhong** [comment]
@CouchofTomato 

Interesting that this PR's latest state doesn't have those href changes.

Currently, we have #28421 in review as part of the accessible link text epic for the Rails pathway, and that PR makes changes to the links in the Active Record Queries lesson. It doesn't change any of the hrefs, only the link text. Do you think it would be sensible to close this PR, and request href changes for the relevant links in that PR?

As well as any additional changes to the link text text regarding which sections should actually be read (e.g. 7 and 21, then 8-11 instead of 7 8 21 and 9-12)?

**Comment by MaoShizhong** [comment]
@CouchofTomato Just an update that the `#group` URL has been fixed to `#grouping` in #28421.

Regarding the required reading sections, could this PR be closed then a new issue opened up for contribution to handle that?

**Comment by CouchofTomato** [comment]
@MaoShizhong Yeah I'm ok with that

**Comment by MaoShizhong** [comment]
@CouchofTomato Then I'll leave the new issue with you, as you'll know best what needs to be changed.

**Pull request closed without merging**
